staging/src/k8s.io/code-generator/generate-groups.sh

```diff
@@ -49,44 +49,44 @@ shift 4
 (
   # To support running this script from anywhere, we have to first cd into this directory
   # so we can install the tools.
-  cd $(dirname "${0}")
-  go install ${GOFLAGS:-} ./cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
+  cd "$(dirname "${0}")"
+  go install ${GOFLAGS:+${GOFLAGS[@]}} ./cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
```

Comment:
Why `${GOFLAGS:-}` --> `${GOFLAGS:+${GOFLAGS[@]}}`?

Is `$GOFLAGS` a proper array?

Reply:
GOFLAGS can also be set by the user calling the make rules IIRC, and is meant to plumb through flags -- an array or string will work with this.

These should have ~identical behavior, but this one explicitly opts into that behavior without looking like forgotten quotes. Disabling the lint for this line and leaving a comment why could also work but will be more verbose anyhow.

Reply:
Hrm. I would personally find a way to disable the lint here, it's much less idiomatic to do this substitution. As someone who writes and reads Bash, I've really never seen this specific construct and I'm not sure using it to circumvent shellcheck is the best way forward

Reply:
I am well aware of parameter expansions. I am very specifically saying that when an author in shell wants to split a variable along word boundaries, they use `${varname}`, they do _not_ use `"${varname:+${varname[@]}}"`.

Reply:
go1.11+ will automatically consume `$GOFLAGS` from the environment.. we don't really need to specify it at all.

That said, parameter substitution would only happen here if GOFLAGS is unset.. so if GOFLAGS is unset, then use GOFLAGS? That doesn't make sense.

Reply:
@cblecker the `:+` substitutes if _set_, not if _unset_. Definitely tricky...

Reply:
> go1.11+ will automatically consume $GOFLAGS from the environment.. we don't really need to specify it at all.

this is the best answer. let's do that.